Lesson26/Kubra/edit-post/script.js

```diff
@@ -2,6 +2,56 @@
 window.addEventListener('load', () => {
 	const url = new URL(window.location.href);
 	const postId = url.searchParams.get('postId');
+
+
+	const heading = document.getElementById('input-title');
+	const paragraph = document.getElementById('input-body');
+	const editPost = document.getElementById('edit-post');
+
+	fetch(`https://jsonplaceholder.typicode.com/posts/${postId}`)
+		.then((response) => response.json())
+		.then((data) => {
+			heading.value = data.title;
+			paragraph.value = data.body;
+		});
+
+	editPost.addEventListener('click', (e) => {
+		e.preventDefault();
+
+		const newHeading = heading.value;
+		const newParagraph = paragraph.value
+
+		fetch(`https://jsonplaceholder.typicode.com/posts/${postId}`, {
+			method: 'PUT',
+			body: JSON.stringify({
+				id: postId,
+				title: newHeading,
+				body: newParagraph,
+			}),
+			headers: {
+				'Content-type': 'application/json; charset=UTF-8',
+			},
+		})
+			.then((response) => response.json())
+			.then((newPost) => {
+				const editedLiItem = document.getElementById('edited-li-item');
+
+				const editedHeading = document.createElement('h2');
+				editedHeading.classList.add('post-title');
+				editedHeading.innerText = newHeading;
+				editedLiItem.appendChild(editedHeading);
+
+				const editedParagraph = document.createElement('p');
+				editedParagraph.classList.add('post-body');
+				editedParagraph.innerText = newParagraph;
+				editedLiItem.appendChild(editedParagraph);
+			})
+			
+			const clearInputTitle = document.getElementById('input-title')
+			clearInputTitle.value = '';
```

Comment:
You may actually want to be able to edit the same post more than once in a row. I'd recommend not to clear the input values and instead have the current version of the text prefilled there. In short, remove the following:
```
const clearInputTitle = document.getElementById('input-title')
			clearInputTitle.value = '';
			const clearInputBody = document.getElementById('input-body')
			clearInputBody.value = '';
```